Amplify/Categories/DataStore/Model/Model.swift

```diff
@@ -30,3 +30,16 @@ public protocol Model: Codable {
     var id: Identifier { get }
 
 }
+
+public protocol JSONModel: Model {
```

Comment:
1. As discussed, let's rename this protocol to `JSONValueAccessible` or similar (JSONValueAddressable? JSONValueHolder? JSONValueProperties?). Basically, we want to promise that properties of any type conforming to this protocol can be retrieved with the `jsonValue(forKey:)` method, and that the return value is a `JSONValue?`.
1. This shouldn't derive from Model; it's a separate contract, and you could imagine any type conforming to it, regardless of whether it's also a model. Conforming types that are also Models can conform to both.
1. Probably belongs in a separate file, since it's not a Model.

Reply:
1. Renamed to JSONValueHolder, and it returns Any?? similar to subscript in Model
2. Removed
3. Moved